libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/NavigationMapRoute.java

```diff
@@ -553,20 +549,19 @@ private void drawWaypointMarkers(@NonNull MapboxMap mapboxMap, @Nullable Drawabl
       mapboxMap.addImage("destinationMarker", bitmap);
 
       waypointLayer = new SymbolLayer(WAYPOINT_LAYER_ID, WAYPOINT_SOURCE_ID).withProperties(
-        PropertyFactory.iconImage(Function.property(
-          "waypoint",
-          categorical(
-            stop("origin", PropertyFactory.iconImage("originMarker")),
-            stop("destination", PropertyFactory.iconImage("destinationMarker"))
+        PropertyFactory.iconImage(match(
+          get("waypoint"),
+          literal("originMarker"), literal("stop"), literal("originMarker"),
+          literal("destination"), literal("destinationMarker")
```

Comment:
This conversion doesn't seem correct:

match(
 get("waypoint"),
 literal("origin"),  literal("originMarker"),
 literal("destination"), literal("destinationMarker"),
 literal("defaultOutputIfAboveDoesntMatch")
)